pkg/framework/machinesets.go

```diff
@@ -47,39 +47,31 @@ func BuildMachineSetParams(client runtimeclient.Client, replicas int) MachineSet
 	Expect(err).NotTo(HaveOccurred())
 	Expect(clusterInfra.Status.InfrastructureName).ShouldNot(BeEmpty())
 
-	msName := RandomString(clusterInfra.Status.InfrastructureName)
+	uid, err := uuid.NewUUID()
+	Expect(err).NotTo(HaveOccurred())
 
 	return MachineSetParams{
-		Name:         msName,
+		Name:         clusterInfra.Status.InfrastructureName,
 		Replicas:     int32(replicas),
 		ProviderSpec: providerSpec,
 		Labels: map[string]string{
-			"mhc.framework.openshift.io": msName,
-			ClusterKey:                   clusterName,
+			"e2e.openshift.io": uid.String(),
+			ClusterKey:         clusterName,
 		},
 	}
 }
 
 // CreateMachineSet creates a new MachineSet resource.
 func CreateMachineSet(c client.Client, params MachineSetParams) (*mapiv1beta1.MachineSet, error) {
-	if params.Labels == nil {
-		params.Labels = make(map[string]string)
-	}
-
-	// TODO(bison): It would be nice to automatically set the Cluster ID / name
-	// in the labels here, but I'm not sure how to easily find it.
-	params.Labels["e2e.openshift.io"] =
-		fmt.Sprintf("%s-%d", params.Name, time.Now().Unix())
```

Comment:
Will this effect other tests? How many of the tests use `CreateMachineSet`?

Reply:
Yes, it will ensure we will never see a race condition, when the CreateMachineSet was called at the same time, so the labels on two machines would be matching, and there will be no good way for tests to find out how many machines the machine et actually spawned

Reply:
Sounds like a good improvement 👍